File: ruby/lib/svix/models/create_stream_events_in.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module Svix
  class CreateStreamEventsIn
    attr_accessor :events
    # Optionally creates a new Stream alongside the events.
    #
    # If the stream id or uid that is used in the path already exists, this argument is ignored.
    attr_accessor :stream

    ALL_FIELD ||= ["events", "stream"].freeze
    private_constant :ALL_FIELD

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail(ArgumentError, "The input argument (attributes) must be a hash in `Svix::CreateStreamEventsIn` new method")
      end

      attributes.each do |k, v|
        unless ALL_FIELD.include?(k.to_s)
          fail(ArgumentError, "The field #{k} is not part of Svix::CreateStreamEventsIn")
        end

        instance_variable_set("@#{k}", v)
        instance_variable_set("@__#{k}_is_defined", true)
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = Hash.new
      attrs["events"] = attributes["events"].map { |v| Svix::EventIn.deserialize(v) }
      attrs["stream"] = Svix::StreamIn.deserialize(attributes["stream"]) if attributes["stream"]
      new(attrs)
    end

    def serialize
      out = Hash.new
      out["events"] = @events.map { |v| v.serialize } if @events
      out["stream"] = Svix::serialize_schema_ref(@stream) if @stream
      out
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end

File: csharp/Svix/Models/CreateStreamEventsIn.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    public class CreateStreamEventsIn
    {
        [JsonProperty("events", Required = Required.Always)]
        public required List<EventIn> Events { get; set; }

        [JsonProperty("stream")]
        public StreamIn? Stream { get; set; } = null;

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class CreateStreamEventsIn {\n");
            sb.Append("  Events: ").Append(Events).Append('\n');
            sb.Append("  Stream: ").Append(Stream).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}
